godot-core/src/builtin/vectors/vector_macros.rs

```diff
@@ -348,139 +363,300 @@ macro_rules! impl_float_vector_component_fns {
+            /// zero).
+            #[inline]
+            pub fn abs(self) -> Self {
+                Self::from_glam(self.to_glam().abs())
             }
 
-            pub fn bezier_interpolate(self, control_1: Self, control_2: Self, end: Self, t: $Scalar) -> Self {
-                $(
-                    let $comp = self.$comp.bezier_interpolate(control_1.$comp, control_2.$comp, end.$comp, t);
-                )*
+            /// Returns a new vector with all components clamped between the components of `min` and `max`.
+            ///
+            /// Panics
+            /// Panics if `min` > `max`, `min` is NaN, or `max` is NaN.
+            #[inline]
+            pub fn clamp(self, min: Self, max: Self) -> Self {
+                Self::from_glam(self.to_glam().clamp(min.to_glam(), max.to_glam()))
+            }
 
-                Self::new($($comp),*)
+            /// Returns the length (magnitude) of this vector.
+            #[inline]
+            pub fn length(self) -> real {
+                // does the same as glam's length() but also works for integer vectors
+                (self.length_squared() as real).sqrt()
             }
 
-            pub fn cubic_interpolate(self, b: Self, pre_a: Self, post_b: Self, weight: $Scalar) -> Self {
-                $(
-                    let $comp = self.$comp.cubic_interpolate(b.$comp, pre_a.$comp, post_b.$comp, weight);
-                )*
+            /// Squared length (squared magnitude) of this vector.
+            ///
+            /// Runs faster than [`Self::length`], so prefer it if you need to compare vectors or need the
+            /// squared distance for some formula.
+            #[inline]
+            pub fn length_squared(self) -> $Scalar {
+                self.to_glam().length_squared()
+            }
+
+            /// Returns a new vector containing the minimum of the two vectors, component-wise.
+            #[inline]
+            pub fn coord_min(self, other: Self) -> Self {
+                self.glam2(&other, |a, b| a.min(b))
+            }
+
+            /// Returns a new vector containing the maximum of the two vectors, component-wise.
+            #[inline]
+            pub fn coord_max(self, other: Self) -> Self {
+                self.glam2(&other, |a, b| a.max(b))
+            }
+
+            /// Returns a new vector with each component set to 1.0 if it's positive, -1 if it's negative, and 0 if it's zero.
+            #[inline]
+            pub fn sign(self) -> Self {
+                Self::from_glam(self.to_glam().signum())
+            }
+
+            /// A new vector with each component snapped to the closest multiple of the corresponding
+            /// component in `step`.
+            #[inline]
+            pub fn snapped(self, step: Self) -> Self {
+                Self::new(
+                    $(
+                        (self.$comp as real).snapped(step.$comp as real) as $Scalar
+                    ),*
+                )
+            }
```

Comment:
Why does this need an `as real` now?

Previous implementation:

https://github.com/godot-rust/gdext/blob/4f9f47f8ba6a9f0103bab62af5436f96244cbad8/godot-core/src/builtin/vectors/vector_macros.rs#L425-L429

Reply:
The previous implementation was only for float vectors. Since `self.$comp` can now also be a `i32` and `snapped` only exists on `real` it needs to be converted.

Reply:
I don't think code deduplication should come at the cost of introducing floating-point operations for integer vectors. This is a potential issue for precision and performance.

Maybe it can be implemented separately? Otherwise omit `snapped` for integer vectors, for now...

Reply:
I don't think its possible to implement it any other way since `snapped` requires division and Godot's implementation is the same. But I can omit it for now.

Btw. while looking at Godot's `snapped` [implementation](https://github.com/godotengine/godot/blob/master/core/math/math_funcs.cpp#L120) I noticed that it produces different results to the `real` [implementation](https://github.com/godot-rust/gdext/blob/master/godot-core/src/builtin/math/float.rs#L120)
 
 `((self / step + 0.5) * step).floor()` vs `(self / step + 0.5).floor() * step)`

Reply:
We can gladly correct our float impl to match Godot's.

Regarding integer impl, [something along this lines](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=550d96bfd68c0c03b92c7c321f002f59)? Didn't test much, so almost certainly bugged, but it should be possible without float impl. We could of course look at existing impls as well...

Reply:
> I don't think its possible to implement it any other way ...

Or im simply not smart enough😅

> Didn't test much, so almost certainly bugged ...

Only bug I found is that negativ steps dont work but can be fixed with `let step = step.abs();`

Is there a place where I should add the `snapped` function for i32? Or should that go in another pull request?

Reply:
>I don't think its possible to implement it any other way since 

Or I'm simply not smart enough😅

>Didn't test much, so almost certainly bugged

Only found that it breaks when steps are negative

Reply:
Ah, are negative steps supported in Godot? 🤔

Reply:
Yes, as far as I can tell if you snap 7 to 2, you get the higher possible result (8 and not 6) but with -2 you get the lower on (6 and not 8).

Reply:
The Godot documentation doesn't mention negative values, but they are definitely supported by design and not by accident. For a long time, I found https://github.com/godotengine/godot/issues/6399 from a time where issue numbers were still 4 digits... 🏹 

Maybe we could test for a few values that the inner function (generated) returns the same result...